reference/consentrequest.yaml

```diff
@@ -0,0 +1,135 @@
+    
+    
+    __Consent-request resource__
+    
+    Application (A) can check the status of their consent-request by doing an HTTP GET request with the consent-request-uuid as a queryparam. Status can be Pending, Accepted or Declined.
+    
+    __Consent resource__ 
+    
+    Application (A) can check which schools have given consent by doing an HTTP GET request to ask if a specific digitalDeliveryId has given consent. Or to retrieve a list of schools that have given consent. 
+    
+    In the other application (B) the currently open consent-requests will be made visible to users with the access role to allow data sharing (ie. security officer). The user allows the consent-request and a consent with requested scopes is created. If the user declines the consent-request no consent is created. In both cases an update-message is (optionally) sent to the registered callbackurl. 
+    The user can also create a consent without prior consent-request. Therefor the application needs to render all registered parties from the SEM ecosystem, for the user to select and create a consent with user selected scopes (subset of the oauthclients scopes). This consent will be available on the API described below.
+
+    __Token for digideliveryId__
+    
+    When requesting a token on the oauth2 token resource using the client_credentials a queryparameter 'digideliveryid' containing the identifier for the school. A tokenrequest for a token used for requesting consent for a new scope does not need to contain a requested scope.
+        
+  contact:
+    email: jorim.van.der.wijngaard@topicus.nl
+tags:
+  - name: consent
+    description: Operations related to data access consent
+  - name: consentrequest
+    description: Operations to request data access (consent)
+paths:
+  /consentrequest:
+    post:
+      description: Allows an application to request access to data of a new school (digideliveryid) and scope.
+      tags:
+        - consentrequest
+      operationId: requestConsent
+      requestBody:
+        description: 'ConsentRequest that contains list of requested scopes'
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/consentrequest"
+      responses:
+        '200':
+          description: OK, returned consentrequest will now contain a id.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/consentrequest'
+        '403':
+          description: FORBIDDEN, consent scope can not be requested with for this client
+      summary: ''
+  /consentrequest/{consentRequestId}:
+    get:
+      description: Resource to check status of requested consent
```

Comment:
So we can pull.
Should we also define the events to deliver the consent changes? If consent is revoked etc it is nice to be noticed asap